*** formatting.php.orig	Fri Jul  8 18:42:09 2022
--- /var/www/html/wp-includes/formatting.php	Fri Jul  8 18:42:15 2022
***************
*** 4360,4377 ****
   *                those in `$protocols`, or if `$url` contains an empty string.
   */
  function esc_url( $url, $protocols = null, $_context = 'display' ) {
  	$original_url = $url;
  
! 	if ( '' === $url ) {
  		return $url;
  	}
  
  	$url = str_replace( ' ', '%20', ltrim( $url ) );
  	$url = preg_replace( '|[^a-z0-9-~+_.?#=!&;,/:%@$\|*\'()\[\]\\x80-\\xff]|i', '', $url );
  
! 	if ( '' === $url ) {
  		return $url;
  	}
  
  	if ( 0 !== stripos( $url, 'mailto:' ) ) {
  		$strip = array( '%0d', '%0a', '%0D', '%0A' );
--- 4360,4377 ----
   *                those in `$protocols`, or if `$url` contains an empty string.
   */
  function esc_url( $url, $protocols = null, $_context = 'display' ) {
  	$original_url = $url;
  
! 	if (!strcmp('', $url)) {
  		return $url;
  	}
  
  	$url = str_replace( ' ', '%20', ltrim( $url ) );
  	$url = preg_replace( '|[^a-z0-9-~+_.?#=!&;,/:%@$\|*\'()\[\]\\x80-\\xff]|i', '', $url );
  
! 	if (!strcmp('', $url)) {
  		return $url;
  	}
  
  	if ( 0 !== stripos( $url, 'mailto:' ) ) {
  		$strip = array( '%0d', '%0a', '%0D', '%0A' );
