*** wp-includes/formatting.php.orig	Fri Dec  9 15:48:21 2022
--- wp-includes/formatting.php	Fri Dec  9 15:49:39 2022
***************
*** 4372,4385 ****
  function esc_url( $url, $protocols = null, $_context = 'display' ) {
  	$original_url = $url;
  
! 	if ( '' === $url ) {
  		return $url;
  	}
  
  	$url = str_replace( ' ', '%20', ltrim( $url ) );
  	$url = preg_replace( '|[^a-z0-9-~+_.?#=!&;,/:%@$\|*\'()\[\]\\x80-\\xff]|i', '', $url );
  
! 	if ( '' === $url ) {
  		return $url;
  	}
  
--- 4372,4387 ----
  function esc_url( $url, $protocols = null, $_context = 'display' ) {
  	$original_url = $url;
  
! 	/* We use strcmp instead of ===, therefore the wpgarlic mocked equality won't be used */
! 	if (!strcmp('', $url)) {
  		return $url;
  	}
  
  	$url = str_replace( ' ', '%20', ltrim( $url ) );
  	$url = preg_replace( '|[^a-z0-9-~+_.?#=!&;,/:%@$\|*\'()\[\]\\x80-\\xff]|i', '', $url );
  
! 	/* We use strcmp instead of ===, therefore the wpgarlic mocked equality won't be used */
! 	if (!strcmp('', $url)) {
  		return $url;
  	}
  
