--- /var/www/html/wp-includes/formatting.php.orig	2023-10-02 09:45:55.275203991 +0000
+++ /var/www/html/wp-includes/formatting.php	2023-10-02 09:50:49.178813420 +0000
@@ -4468,18 +4468,20 @@
  *                those in `$protocols`, or if `$url` contains an empty string.
  */
 function esc_url( $url, $protocols = null, $_context = 'display' ) {
 	$original_url = $url;

-	if ( '' === $url ) {
+	/* We use strcmp instead of ===, therefore the wpgarlic mocked equality won't be used */
+	if (!strcmp('', $url)) {
 		return $url;
 	}

 	$url = str_replace( ' ', '%20', ltrim( $url ) );
 	$url = preg_replace( '|[^a-z0-9-~+_.?#=!&;,/:%@$\|*\'()\[\]\\x80-\\xff]|i', '', $url );

-	if ( '' === $url ) {
+	/* We use strcmp instead of ===, therefore the wpgarlic mocked equality won't be used */
+	if (!strcmp('', $url)) {
 		return $url;
 	}

 	if ( 0 !== stripos( $url, 'mailto:' ) ) {
 		$strip = array( '%0d', '%0a', '%0D', '%0A' );
