*** /var/www/html/wp-load.php.orig	Fri Jul  8 18:38:42 2022
--- /var/www/html/wp-load.php	Fri Jul  8 18:38:42 2022
***************
*** 59,71 ****
  	// A config file doesn't exist.
  
  	define( 'WPINC', 'wp-includes' );
  	require_once ABSPATH . WPINC . '/load.php';
  
- 	// Standardize $_SERVER variables across setups.
- 	wp_fix_server_vars();
- 
  	require_once ABSPATH . WPINC . '/functions.php';
  
  	$path = wp_guess_url() . '/wp-admin/setup-config.php';
  
  	/*
--- 59,68 ----
