*** /var/www/html/wp-settings.php.orig	Fri Jul  8 18:38:42 2022
--- /var/www/html/wp-settings.php	Fri Jul  8 18:38:42 2022
***************
*** 65,77 ****
  
  // WordPress calculates offsets from UTC.
  // phpcs:ignore WordPress.DateTime.RestrictedFunctions.timezone_change_date_default_timezone_set
  date_default_timezone_set( 'UTC' );
  
- // Standardize $_SERVER variables across setups.
- wp_fix_server_vars();
- 
  // Check if we're in maintenance mode.
  wp_maintenance();
  
  // Start loading timer.
  timer_start();
--- 65,74 ----
***************
*** 480,492 ****
  do_action( 'plugins_loaded' );
  
  // Define constants which affect functionality if not already defined.
  wp_functionality_constants();
  
- // Add magic quotes and set up $_REQUEST ( $_GET + $_POST ).
- wp_magic_quotes();
- 
  /**
   * Fires when comment cookies are sanitized.
   *
   * @since 2.0.11
   */
--- 477,486 ----
