--- wp-settings.php.orig	2023-10-02 09:45:51.435209243 +0000
+++ wp-settings.php	2023-10-02 09:45:51.443209232 +0000
@@ -495,13 +492,10 @@
 do_action( 'plugins_loaded' );

 // Define constants which affect functionality if not already defined.
 wp_functionality_constants();

-// Add magic quotes and set up $_REQUEST ( $_GET + $_POST ).
-wp_magic_quotes();
-
 /**
  * Fires when comment cookies are sanitized.
  *
  * @since 2.0.11
  */
